src/jit/gentree.h

```diff
@@ -4097,6 +4101,30 @@ inline bool GenTree::IsZero()
     return false;
 }
 
+// Note that, unlike IsZero(), this only returns true only for integral types.
+inline bool GenTree::IsIOne()
+{
+    if ((gtOper == GT_CNS_INT) && (gtIntConCommon.IconValue() == 1))
+        return true;
+
+    if ((gtOper == GT_CNS_LNG) && (gtIntConCommon.LngValue() == 1))
+        return true;
+
```

Comment:
It would be confusing if this method is defined on GenTree and doesn't handle floating-point constants.  I would suggest to do one of the following:

1) Rename the method to IsIntegralOne() to be indicative of its purpose
2) Add handling for floating point constants as well
e.g.
`if ((gtOper == GT_CNS_DBL) && (gtDblCon.gtDconVal == 1.0))`